src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -392,24 +428,20 @@ internal CompilationWithAnalyzers GetCompilationWithAnalyzers(Project project, C
             }
         }
 
-        private bool SkipRunningAnalyzer(
-            CompilationOptions compilationOptions,
-            DiagnosticAnalyzerDriver userDiagnosticDriver,
-            bool openedDocument,
-            bool skipClosedFileChecks,
-            StateSet stateSet)
+        private async Task<bool> SkipRunningAnalyzerAsync(Project project, DiagnosticAnalyzer analyzer, bool openedDocument, bool skipClosedFileCheck, CancellationToken cancellationToken)
         {
-            if (Owner.IsAnalyzerSuppressed(stateSet.Analyzer, userDiagnosticDriver.Project))
+            // this project is not in a good state. only continue if it is opened document.
+            if (!await project.HasSuccessfullyLoadedAsync(cancellationToken).ConfigureAwait(false) && !openedDocument)
```

Comment:
that is what it does.

if (await SkipRunningAnalyzerAsync(document.Project, stateSet.Analyzer, openedDocument, skipClosedFileCheck: false, cancellationToken: cancellationToken).ConfigureAwait(false))
                    {
                        await ClearExistingDiagnostics(document, stateSet, StateType.Syntax, cancellationToken).ConfigureAwait(false);
                        continue;
                    }